cmd/node-termination-handler.go

```diff
@@ -242,6 +242,7 @@ func main() {
 					log.Info().
 						Str("event-id", event.EventID).
 						Str("kind", event.Kind).
+						Str("monitor", event.Monitor).
 						Str("node-name", event.NodeName).
 						Str("instance-id", event.InstanceID).
 						Str("provider-id", event.ProviderID).
```

Comment:
On line 252 when we emit the Kubernetes event, we need to keep the "reason" argument the same as it was before (`observability.GetReasonForKind(event.Kind)`).

However, let's add another argument at the end, after `event.Description` that includes the `AWSEventDetails`. I'd suggest we use the "screaming snake case" names without conversion, but I'm open to doing a conversion like `GetReasonForKind()` does right now if you disagree.

Reply:
Will do, but I have to respectfully say I'm against this one. In all other cases the old and wrong field, data, log line or whatever can be kept and coexist with a new and fixed one. In this particular case like you pointed out there's only one reason, so either we use the right or the wrong reason. Kubernetes event reasons are the cornerstone of event-based Kubernetes monitoring, and we're choosing to leave it wrong. I understand backwards compatibility but in this case here we're not talking about an enhancement or a new feature but a fix, and I think not fixing things that are known to be wrong is not the best way to go.

That said, I think the best way to go is to leave the `recorder.Emit()` call in 252 as is and instead modify the `Description` field generation in all monitors. I.e., we would change line 93 in `spot-itn-monitor.go` to something like
```
Description:  fmt.Sprintf("Spot ITN received. Instance will be interrupted at %s. AWS event type: %s \n", instanceAction.Time, monitor.SpotITNKind),
```
or maybe
```
Description:  fmt.Sprintf("Spot ITN received (%s). Instance will be interrupted at %s \n", monitor.SpotITNKind, instanceAction.Time),
```
WDYT?

Reply:
Are you also intending to keep the return values of `observability.GetReasonForKind()` as they were before, such that the only change to the argument values in `recorder.Emit()` would be this updated `Description` field? If so, I am in favor of that plan.

I like your first suggestion (`AWS event type: ...` at the end of the `Description`).

Reply:
Yes, if we are to keep the old and wrong reason there is no other option than reverting the changes in `GetReasonForKind()` so they return the old values.